### DISCUSSION TITLE
Clarity AI Upscaler & Enhancer Reproduction

### DISCUSSION THREAD

**Discussion by deltheil** [finegrain-ai/refiners#393] [Announcements]
# tl;dr

A few months ago, [philz1337x](https://x.com/philz1337x/status/1768679154726359128) launched [Clarity Upscaler](https://github.com/philz1337x/clarity-upscaler), an open-source tool built upon Stable Diffusion 1.5 and additional techniques (ControlNet Tile, ...) for advanced upscaling and enhancement. Inspired by its performance, we've re-implemented it with the Refiners library to offer a standalone Python version, easily customizable for various projects.

# Dive in!

- 🧪 Try it out on our [HuggingFace Space](https://huggingface.co/spaces/finegrain/enhancer)
- 📖 Explore the details in the [blog post](https://blog.finegrain.ai/posts/reproducing-clarity-upscaler/)
- ⚙️ Dive into the codebase, e.g. [MultiUpscaler base class](https://github.com/finegrain-ai/refiners/blob/09a9dfd494f06d0cb0af8b0c78ea3460b9180cd1/src/refiners/foundationals/latent_diffusion/stable_diffusion_1/multi_upscaler.py#L61) and [ESRGANUpscaler example](https://huggingface.co/spaces/finegrain/enhancer/blob/13531f3129760bfa8cb413656d57ea48b1204baf/src/enhancer.py#L21)

cc @holwech
